src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp

```diff
@@ -42,77 +42,85 @@ FleetApiClient::FleetApiClient(
 }
 
 
-void FleetApiClient::setDeviceIdentification(const fleet_protocol::cxx::DeviceID &deviceId) {
+void FleetApiClient::setDeviceIdentification(const cxx::DeviceID &deviceId) const {
 	auto deviceIdentification = deviceId.getDeviceId();
 	deviceIdPtr_->setModuleId(deviceIdentification.module);
 	deviceIdPtr_->setType(deviceIdentification.device_type);
 	deviceIdPtr_->setRole(std::string(static_cast<char*>(deviceIdentification.device_role.data)));
 	deviceIdPtr_->setName(std::string(static_cast<char*>(deviceIdentification.device_name.data)));
 }
 
-
-std::vector<std::shared_ptr<model::Car>> FleetApiClient::getCars(std::optional<int64_t> since, std::optional<bool> wait) {
-	auto carsRequest = carApi_->availableCars(wait.value_or(false), since.value_or(0));
+std::pair<std::vector<std::shared_ptr<model::Car>>, bool> FleetApiClient::getCars(
```

Comment:
Please do not use Bool. Use enum class. What if another state needs to be returned?

Maybe create one Enum class for the FleetApiClient...

Reply:
Enum for return codes added